samples/MemoryCacheSample/Program.cs

```diff
@@ -30,89 +33,73 @@ public void Main()
-            result = cache.GetOrSet(key, context => new object());
-
-            // Retrieve / Create when we want to lazily create the object.
-            result = cache.GetOrSet(key, state, context => new object());
-
             // Cache entry configuration:
 
             // Stays in the cache as long as possible
-            result = cache.GetOrSet(key, state, context =>
-            {
-                context.SetPriority(CachePreservationPriority.NeverRemove);
-                return new object();
-            });
+            result = cache.Set(
+                key,
+                new object(),
+                new CacheEntryOptions().SetPriority(CacheItemPriority.NeverRemove));
 
             // Automatically remove if not accessed in the given time
-            result = cache.GetOrSet(key, state, context =>
-            {
-                context.SetSlidingExpiration(TimeSpan.FromMinutes(5));
-                return new object();
-            });
+            result = cache.Set(
+                key,
+                new object(),
+                new CacheEntryOptions().SetSlidingExpiration(TimeSpan.FromMinutes(5)));
 
             // Automatically remove at a certain time
-            result = cache.GetOrSet(key, state, context =>
-            {
-                context.SetAbsoluteExpiration(new DateTime(2014, 12, 31));
-                // or relative:
-                // context.SetAbsoluteExpiration(TimeSpan.FromMinutes(5));
-                return new object();
-            });
+            result = cache.Set(
+                key,
+                new object(),
+                new CacheEntryOptions().SetAbsoluteExpiration(DateTimeOffset.UtcNow.AddDays(2)));
 
             // Automatically remove if not accessed in the given time
             // Automatically remove at a certain time (if it lives that long)
-            result = cache.GetOrSet(key, state, context =>
-            {
-                context.SetSlidingExpiration(TimeSpan.FromMinutes(5));
-
-                context.SetAbsoluteExpiration(new DateTime(2014, 12, 31));
-                // or relative:
-                // context.SetAbsoluteExpiration(TimeSpan.FromMinutes(5));
-                return new object();
-            });
+            result = cache.Set(
+                key,
+                new object(),
+                new CacheEntryOptions()
+                .SetSlidingExpiration(TimeSpan.FromMinutes(5))
+                .SetAbsoluteExpiration(DateTimeOffset.UtcNow.AddDays(2)));
 
             // Callback when evicted
-            result = cache.GetOrSet(key, state, context =>
-            {
-                context.RegisterPostEvictionCallback((echoKey, value, reason, substate) =>
-                    Console.WriteLine(echoKey + ": '" + value + "' was evicted due to " + reason), state: null);
-                return new object();
-            });
+            result = cache.Set(
+                key,
```

Comment:
Can you have an example of a set call that doesn't chain the options into a single argument?

```
var options  = new CacheEntryOptions()
                                .RegisterPostEvictionCallback((echoKey, value, reason, substate) => 
                                     Console.WriteLine(echoKey + ": '" + value + "' was evicted due to " + reason), state: null);

resukt = cache.Set(key, new object(), options);
```

But with better formatting